docs/user-guide/lrustorecache.rst

```diff
@@ -0,0 +1,186 @@
+.. only:: doctest
+
+   >>> import shutil
+   >>> shutil.rmtree('test.zarr', ignore_errors=True)
+
+.. _user-guide-lrustorecache:
+
+LRUStoreCache guide
+===================
+
+The :class:`zarr.storage.LRUStoreCache` provides a least-recently-used (LRU) cache layer
+that can be wrapped around any Zarr store to improve performance for repeated data access.
+This is particularly useful when working with remote stores (e.g., S3, HTTP) where network
+latency can significantly impact data access speed.
+
+The LRUStoreCache implements a cache that stores frequently accessed data chunks in memory,
+automatically evicting the least recently used items when the cache reaches its maximum size.
+
+.. note::
+   The LRUStoreCache is a wrapper store that maintains compatibility with the full
+   :class:`zarr.abc.store.Store` API while adding transparent caching functionality.
+
+Basic Usage
+-----------
+
+Creating an LRUStoreCache is straightforward - simply wrap any existing store with the cache:
+
+   >>> import zarr
+   >>> import zarr.storage
+   >>> import numpy as np
+   >>>
+   >>> # Create a local store and wrap it with LRU cache
+   >>> local_store = zarr.storage.LocalStore('test.zarr')
+   >>> cache = zarr.storage.LRUStoreCache(local_store, max_size=1024 * 1024 * 256)  # 256MB cache
+   >>>
+   >>> # Create an array using the cached store
+   >>> zarr_array = zarr.zeros((100, 100), chunks=(10, 10), dtype='f8', store=cache, mode='w')
+   >>>
+   >>> # Write some data to force chunk creation
+   >>> zarr_array[:] = np.random.random((100, 100))
+
+The ``max_size`` parameter controls the maximum memory usage of the cache in bytes. Set it to
+``None`` for unlimited cache size (use with caution).
+
+Performance Benefits
+-------------------
```

Comment:
There are still some titles that don't have matching underline lengths in this file.